ocean_provider/util_url.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import ipaddress
+import dns.resolver
+from ocean_provider.utils.basics import get_config
+
+from urllib.parse import urlparse
+
+logger = logging.getLogger(__name__)
+
+
+def is_safe_url(url):
+    if not is_safe_schema(url):
+        return False
+    result = urlparse(url)
+    return is_safe_domain(result.netloc)
+
+
+def is_safe_schema(url):
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc, result.path])
+    except:
```

Comment:
can we identify an exception class here, or a list of exception classes to be caught?

Reply:
do we need to ? after all, we need to make sure that a URL is valid (has a schema, a netloc (domain),and a path.
If it doesn't, we will reject it anyway

Reply:
I mean the `except :` thing is not usually recommended.  The fewer of these the better?